scripts/execute_cypress.sh

```diff
@@ -0,0 +1,190 @@
+    echo "${service}-${connector}" >> "${tmp_file}"
+  fi
+}
+export -f execute_test
+
+# Function to run tests
+function run_tests() {
+  local jobs="${1:-1}"
+  tmp_file=$(mktemp)
+
+  # Ensure temporary file is removed on script exit
+  trap 'cleanup' EXIT
+
+  for service in "${connector_map[@]}"; do
+    declare -n connectors="$service"
+
+    if [[ ${#connectors[@]} -eq 0 ]]; then
+      # Service-level test (e.g., payment-method-list or routing)
+      [[ $service == "payment_method_list" ]] && service="payment-method-list"
+
+      echo "Running ${service} tests without connectors..."
+      export REPORT_NAME="${service}_report"
+
+      if ! npm run "cypress:${service}"; then
+        echo "${service}" >> "${tmp_file}"
+      fi
+    else
+      # Connector-specific tests (e.g., payments or payouts)
+      print_color "yellow" "Running tests for service: '${service}' with connectors: [${connectors[*]}] in batches of ${jobs}..."
+
+      # Execute tests in parallel
+      printf '%s\n' "${connectors[@]}" | parallel --jobs "${jobs}" execute_test {} "${service}" "${tmp_file}"
+    fi
+  done
+
+  # Collect failed connectors
+  if [[ -s "${tmp_file}" ]]; then
+    failed_connectors=($(< "${tmp_file}"))
+    print_color "red" "One or more connectors failed to run:"
+    printf '%s\n' "${failed_connectors[@]}"
+    exit 1
+  else
+    print_color "green" "Cypress tests execution successful!"
+  fi
+}
+
+# Function to check and install dependencies
+function check_dependencies() {
+  # parallel and npm are mandatory dependencies. exit the script if not found.
+  local dependencies=("parallel" "npm")
+
+  for cmd in "${dependencies[@]}"; do
+    if ! command_exists "$cmd"; then
+      print_color "red" "ERROR: ${cmd^} is not installed!"
+      exit 1
+    else
+      print_color "green" "${cmd^} is installed already!"
+
+      if [[ ${cmd} == "npm" ]]; then
+        npm ci || {
+          print_color "red" "Command \`npm ci\` failed!"
+          exit 1
+        }
+      fi
+    fi
+  done
+}
+
+# Cleanup function to handle exit
+function cleanup() {
+  print_color "yellow" "Cleaning up..."
+  if [[ -d "cypress-tests" ]]; then
+    cd - || return
+  fi
+
+  if [[ -n "${tmp_file}" && -f "${tmp_file}" ]]; then
+    rm -f "${tmp_file}"
+  fi
+}
+
+# Main function
+function main() {
+  local command="${1:-}"
+  local jobs="${2:-4}"
+
+  # Ensure script runs from 'cypress-tests' directory
+  if [[ "$(basename "$PWD")" != "cypress-tests" ]]; then
+    print_color "yellow" "Changing directory to 'cypress-tests'..."
+    cd cypress-tests || {
+      print_color "red" "ERROR: Directory 'cypress-tests' not found!"
+      exit 1
+    }
+  fi
+
+  check_dependencies
+  read_service_arrays
+
+  case "$command" in
+    --parallel | -p)
+      print_color "yellow" "WARNING: Running Cypress tests in parallel is more resource-intensive!"
+      # At present, parallel execution is limited to batch of 4 by default to not run out of memory
```

Comment:
The number `4` is specific to our CI setup for now. Can we avoid documenting that number, which is specific to our current CI setup in this script, since the script should not depend on the resources available in our CI setup?

Reply:
i had mentioned `4` because in the main function, i've set `4` as default.

anyway, removed.